### PR TITLE
Split tf2_ros in tf2_ros and tf2_ros_py

### PR DESCRIPTION
**Pull request by ahcorde**
Following this issue  https://github.com/ros2/geometry2/issues/169 I splitted `tf2_ros` in 2 packages:

 - tf2_ros
 - tf2_ros_py

### CODE REVIEW

File: tf2_ros/include/tf2_ros/message_filter.h [hunk 1]
```diff
@@ -448,12 +448,9 @@ class MessageFilter : public MessageFilterBase, public message_filters::SimpleFi
   void add(const MConstPtr & message)
   {
     using builtin_interfaces::msg::Time;
-    std::shared_ptr<std::map<std::string, std::string>> header(new std::map<std::string,
-      std::string>);
 
-    (*header)["callerid"] = "unknown";
     Time t = node_clock_->get_clock()->now();
-    add(MEvent(message, header, t));
+    add(MEvent(message, t));
```
Comment (clalancette) [inline tf2_ros/include/tf2_ros/message_filter.h]: This seems to be an unrelated change; can you split this out separately?

File: tf2_ros_py/package.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  <depend>tf2</depend>
+  <depend>tf2_msgs</depend>
+  <depend>tf2_py</depend>
+  <depend>tf2_ros</depend>
+
+  <!-- TODO(tfoote) dependencies from unported sections restore when ported
+  <depend>xmlrpcpp</depend> -->
+  <!-- <build_depend>rosgraph</build_depend> -->
+
+  <!-- <test_depend>rostest</test_depend> -->
```
Comment (clalancette) [inline tf2_ros_py/package.xml]: I think we can just remove these at this point.  `xmlrpcpp` will never be a thing in ROS 2, the testing infrastructure is different (rostest goes away), and we'll reintroduce rosgraph if we need it.

File: tf2_ros_py/test/test_buffer_client.py [hunk 3]
```diff
@@ -192,7 +192,7 @@ def test_lookup_transform_true(self):
             self.node, 'lookup_transform', check_frequency=10.0, timeout_padding=0.0)
 
         result = buffer_client.lookup_transform(
-            'foo', 'bar', rclpy.time.Time(), rclpy.duration.Duration(seconds=5.0))
+            'foo', 'bar', rclpy.time.Time().to_msg(), rclpy.duration.Duration(seconds=5.0))
```
Comment (clalancette) [inline tf2_ros_py/test/test_buffer_client.py:132]: Why is this change (and the similar ones below) needed in this port, but not in the original?

**Review by clalancette — CHANGES_REQUESTED** [review]
This generally looks good to me, but I'll request that we do unrelated cleanup in separate PRs.  That way this PR is just code motion, no changes at all.

File: tf2_ros/package.xml [hunk 4]
```diff
@@ -30,12 +30,6 @@
   <depend>tf2_msgs</depend>
   <depend>tf2_py</depend>
```
Comment (clalancette) [inline tf2_ros/package.xml]: GitHub UI won't let me leave comments above this line, but this whole package.xml needs to be cleaned up.

1.  Now that we have no python things in here, the only "buildtool_depend" we need is `ament_cmake`.
1.  There is no direct dependency in this repository on `std_msgs`, so it should be removed.
1.  We may as well make `geometry_msgs` a regular `<depend>`, like the rest of the dependencies.
1.  This package no longer has a dependency on tf2_py.
1.  This package no longer needs pytest as a `test_depend`

Similar cleanups should probably be done in the CMakeLists.txt

File: tf2_ros_py/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.5)
```
Comment (clalancette) [inline tf2_ros_py/CMakeLists.txt]: Is there any reason we can't make this a pure python package (no CMakeLists.txt at all)?  I think that would be more pythonic.

File: tf2_ros_py/package.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  <depend>message_filters</depend>
+  <depend>rclcpp</depend>
+  <depend>rclcpp_action</depend>
+  <depend>rclcpp_components</depend>
+  <depend>rclpy</depend>
+  <depend>tf2</depend>
+  <depend>tf2_msgs</depend>
+  <depend>tf2_py</depend>
+  <depend>tf2_ros</depend>
```
Comment (clalancette) [inline tf2_ros_py/package.xml]: For a python package, there are really no `<build_depend>`, so just make all of these `<exec_depend>`

File: tf2_ros_py/package.xml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  <author>Eitan Marder-Eppstein</author>
+  <author>Wim Meeussen</author>
+  <maintainer email="tfoote@osrfoundation.org">Tully Foote</maintainer>
+  <license>BSD</license>
+
+  <url type="website">http://www.ros.org/wiki/tf2_ros</url>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+  <buildtool_depend>ament_cmake_python</buildtool_depend>
+  <buildtool_depend>python_cmake_module</buildtool_depend>
```
Comment (clalancette) [inline tf2_ros_py/package.xml]: If we end up making this a pure Python package, these will go away.

File: tf2_ros_py/test/test_buffer_client.py (hunk 3 above)
Comment (clalancette) [reply tf2_ros_py/test/test_buffer_client.py:132]: This comment is still open (and there are a couple more instances of it below).

**Review by clalancette — CHANGES_REQUESTED** [review]
A bunch of changes in here.  The biggest one is switching tf2_ros_py to a pure Python package.

**Comment by clalancette** [comment]
All right, I reworked the tests here to succeed, and I've converted this to a pure Python package.  I'm going to do a self-review and run CI, but @ahcorde it would be good if you could take a look here.

File: tf2_ros_py/test/test_buffer_client.py (hunk 3 above)
Comment (clalancette) [reply tf2_ros_py/test/test_buffer_client.py:132]: That's just the way it works now.  The original didn't work.  Resolving this conversation.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11068/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6381/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9044/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11007/)

**Review by clalancette — APPROVED** [review]
I'm going to approve this now, but this could use another look since I did some non-trivial changes here.

File: tf2_ros_py/tf2_ros/buffer_client.py [hunk 8]
```diff
@@ -65,36 +65,36 @@ def __init__(self, node, ns, check_frequency = 10.0, timeout_padding = Duration(
...
 
         return self.__process_goal(goal)
 
-    # lookup, advanced api 
+    # lookup, advanced api
     def lookup_transform_full(self, target_frame, target_time, source_frame, source_time, fixed_frame, timeout=Duration()):
         """
         Get the transform from the source frame to the target frame using the advanced API.
 
         :param target_frame: Name of the frame to transform into.
```
Comment (ahcorde) [inline tf2_ros_py/tf2_ros/buffer_client.py:94]: :type *: is not defined as well as :rtype: . Should we open a ticket?

**Review by ahcorde — COMMENTED** [review]
LGTM, thank you @clalancette to take over this PR

File: tf2_ros_py/tf2_ros/buffer_client.py (hunk 8 above)
Comment (clalancette) [reply tf2_ros_py/tf2_ros/buffer_client.py:94]: Yeah, good call.  I'll open an issue to follow-up and add those annotations.

**Comment by clalancette** [comment]
All right, CI was green, added a note to the release page.  Merging in, thanks for the review.

**Pull request merged**